gadget/update.go

```diff
@@ -1440,21 +1535,54 @@ type Updater interface {
 	Rollback() error
 }
 
-func applyUpdates(new GadgetData, updates []updatePair, rollbackDir string, observer ContentUpdateObserver) error {
+func updateLocationForStructure(structureLocations map[string]map[int]StructureLocation, ps *LaidOutStructure) (loc StructureLocation, err error) {
+	if !ps.HasFilesystem() {
+		loc, ok := structureLocations[ps.VolumeName][ps.YamlIndex]
+		if !ok {
+			return loc, fmt.Errorf("structure with index %d on volume %s not found", ps.YamlIndex, ps.VolumeName)
+		}
```

Comment:
This can be taken out of the `if`, because it's the same as in the `else` (unless I'm missing something)

Reply:
done